### ISSUE TITLE
Support WSL pkg managers (such as APT)

### ISSUE THREAD

**Issue by ppvnf** [marticliment/WingetUI#1490]
> This is the same proposal as #1241, right?

possibly, but here he mentions snap and flatpak. 
Maybe an integration with WSL would be nice as well

_Originally posted by @ppvnf in https://github.com/marticliment/WingetUI/issues/1307#issuecomment-1756842333_

**Comment by MrStanDu33** [comment]
I think it could be nice to access all package managers in each distros from this tool !

Not sure tho how it could be implemented in the UI itself 🫤

**Comment by Andrew-J-Larson** [comment]
> I think it could be nice to access all package managers in each distros from this tool !
> 
> Not sure tho how it could be implemented in the UI itself 🫤

1. Get list of available WSL distros (`wsl -l`)
2. `For` each distro, check for all available package managers (e.g. `apt`, `pac`, `snap`, `flatpak`, etc.)... and/or in the future use `vso` (originally `apx` in Vanilla OS 1.0) since Vanilla OS plans to make `vso` a distro independent all-in-one package manager, which would cut down a lot of work that would be needed for Winget UI)
  a. If **not** using `vso`, it would need to manually gather lists from each package manager and execute their commands.
  b. If using `vso`, checking updates, and updating would be very simple to implement. The only little downside is a one-time install of VSO into each WSL distro.

**Comment by Andrew-J-Larson** [comment]
Yep, `apx` appears to be stable at version 2.3.0 at time of writing, so we should probably look at integrating support for that through Winget UI

**Comment by Andrew-J-Larson** [comment]
Of course though, would need to include a warning about installing `vso` in each distro... and probably want more delicate control for any arch based distros, since you don't always want to upgrade everything there...

**Comment by Andrew-J-Larson** [comment]
The instructions for using `apx` on other distros is available on its source page:
- https://github.com/Vanilla-OS/apx?tab=readme-ov-file#instructions-for-using-apx-in-other-distributions

**Comment by Andrew-J-Larson** [comment]
Ah wait, reading more into `apx`, just realized it would just manage specific subsystems, but either way most of the work is done, just would have to fork the software, make changes so that interacts with the distro that it's in (rather than an assumed subsystem), and then things should work just fine... (but if someone has vanilla os as a WSL distro, or `apx` already installed in one of their distros, probably wouldn't want to overwrite it).

**Comment by Andrew-J-Larson** [comment]
Another option might be **[`upm`](https://github.com/epitron/upm)**

**Comment by marticliment** [comment]
I will take this suggestions into account

**Comment by rnett** [comment]
Even if distro package managers aren't included (or aren't included initially), supporting ecosystem package managers that are already supported on windows (npm, cargo, pip, etc) would still be extremely useful.

**Comment by sduquette-devolutions** [comment]
We have not officially announced it yet but we now build a Linux (and MacOS) version of UniGetUI. We also added support for many Linux package managers including apt, dnf, pacman, flatpak and snapcraft. And UniGetUI works from WSL.

https://github.com/Devolutions/UniGetUI/releases/tag/v2026.1.8

I am closing this issue as the original request is implemented. Please open new issues if you encounter problems or if want to request package managers that we are not currently supporting. 

<img width="1452" height="787" alt="Image" src="https://github.com/user-attachments/assets/1a14e9e9-30ac-4148-a08d-8fc5750ea5b8" />

**Comment by rnett** [comment]
I can't speak for the others in this thread, but this isn't what I was looking for.  I want to run one app, in Windows, and be able to manage both Windows and WSL deps.  This seems like a large step in the that direction but still requires using and managing two separate apps.  Should I create a separate issue for that?

**Comment by sduquette-devolutions** [comment]
Thanks for your response Ryan, I reread the thread and other participants requested the same thing. I changed the issue title and reopened it.

**Comment by dominch** [comment]
hello, I found this thread and I'm looking at same thing. 
for now I need to manually update my debian sitting in wsl, just by login as root and make apt update && apt upgrade (or with nala command, bit nicer apt).
It would be awesome to get this functionality in UniGetUi, my wsl is usually up and running in background, so it's quite easy to detect it and try to get packages list. Looking forward for this here.

**Comment by dominch** [comment]
any update regarding this?
